tests/views/test_error_handlers.py

```diff
@@ -0,0 +1,32 @@
+from unittest import mock
+
+from application.utils.exceptions import DatabaseError
+from tests.test_client import TestClient
+
+
+class TestExceptionHandlers(TestClient):
+
+    @mock.patch('application.views.classification_type_view.classification_type_controller')
+    def test_handle_exception(self, mock_classification_type_controller):
+        # Given an unhandled exception
+        mock_classification_type_controller.delete_classification_type.side_effect = Exception
+
+        # When we delete a classification type
+        response = self.client.delete('/classificationtype/LEGAL_BASIS', headers=self.get_auth_headers())
+
+        # we get an error response
+        self.assertStatus(response, 500)
+        self.assertEquals(response.json, {'error': "Internal Server error"})
+
+    @mock.patch('application.views.classification_type_view.classification_type_controller')
+    def test_handle_database_exception(self, mock_classification_type_controller):
+        # Given the database throws an exception
+        mock_classification_type_controller.delete_classification_type.side_effect = DatabaseError(
+            error="Database Exception", status_code=400)
+
+        # When we delete a classification type
+        response = self.client.delete('/classificationtype/LEGAL_BASIS', headers=self.get_auth_headers())
+
+        # we get an appropriate error response
+        self.assertStatus(response, 400)
```

Comment:
Might be worth changing the status_code to 500 as it through me when I first read a database exception raises a 400 until I realised it was being mocked

Reply:
I've changed it from the default 500 just to check that it maps the internal exception status code across.

Reply:
Will a database error ever be anything but 500 though?